egs/wsj/s5/steps/nnet3/train_discriminative.sh

```diff
@@ -0,0 +1,376 @@
+          $dir/$[$x+1].$n.raw || touch $dir/.error &
+      done
+      wait
+      [ -f $dir/.error ] && exit 1
+    )
+    [ -f $dir/.error ] && { echo "Found $dir/.error. See $dir/log/train.$x.*.log"; exit 1; }
+
+    nnets_list=$(for n in $(seq $num_jobs_nnet); do echo $dir/$[$x+1].$n.raw; done)
+
+    # below use run.pl instead of a generic $cmd for these very quick stages,
+    # so that we don't run the risk of waiting for a possibly hard-to-get GPU.
+    run.pl $dir/log/average.$x.log \
+      nnet3-average $nnets_list - \| \
+      nnet3-am-copy --set-raw-nnet=- $dir/$x.mdl $dir/$[$x+1].mdl || exit 1;
+
+    if $modify_learning_rates; then
+      run.pl $dir/log/modify_learning_rates.$x.log \
+        nnet3-modify-learning-rates --retroactive=$retroactive \
+        --last-layer-factor=$last_layer_factor \
+        --first-layer-factor=$first_layer_factor \
+        "nnet3-am-copy --raw $dir/$x.mdl -|" "nnet3-am-copy --raw $dir/$[$x+1].mdl -|" - \| \
+        nnet3-am-copy --set-raw-nnet=- $dir/$x.mdl $dir/$[$x+1].mdl || exit 1;
+    fi
+    rm $nnets_list
+
+    if [ ! -z "${iter_to_epoch[$x]}" ]; then
+      e=${iter_to_epoch[$x]}
+      ln -sf $x.mdl $dir/epoch$e.mdl
+    fi
+
+    if $adjust_priors && [ ! -z "${iter_to_epoch[$x]}" ]; then
+      if [ ! -f $degs_dir/priors_egs.1.ark ]; then
+        echo "$0: Expecting $degs_dir/priors_egs.1.ark to exist since --adjust-priors was true."
+        echo "$0: Run this script with --adjust-priors false to not adjust priors"
+        exit 1
+      fi
+      (
+      e=${iter_to_epoch[$x]}
+      rm -f $dir/.error 2> /dev/null || true
+
+      steps/nnet3/adjust_priors.sh --egs-type priors_egs \
+        --num-jobs-compute-prior $num_archives_priors \
+        --cmd "$cmd $prior_queue_opt" --use-gpu false \
+        --raw false --iter epoch$e $dir $degs_dir \
+        || { touch $dir/.error; echo "Error in adjusting priors. See $dir/log/adjust_priors.epoch$e.log"; exit 1; }
+      ) &
+    fi
+
+    [ -f $dir/.error ] && { echo "Found $dir/.error. Error on iteration $x"; exit 1; }
+  fi
+
+  x=$[$x+1]
+  num_archives_processed=$[num_archives_processed+num_jobs_nnet]
+done
+
+rm -f $dir/final.mdl 2>/dev/null || true
+cp $dir/$x.mdl $dir/final.mdl
+ln -sf final.mdl $dir/epoch$[num_epochs*frame_subsampling_factor].mdl
+
+if $adjust_priors && [ $stage -le $num_iters ]; then
+  if [ ! -f $degs_dir/priors_egs.1.ark ]; then
+    echo "$0: Expecting $degs_dir/priors_egs.1.ark to exist since --adjust-priors was true."
+    echo "$0: Run this script with --adjust-priors false to not adjust priors"
+    exit 1
+  fi
+
+  steps/nnet3/adjust_priors.sh --egs-type priors_egs \
+    --num-jobs-compute-prior $num_archives_priors \
+    --cmd "$cmd $prior_queue_opt" --use-gpu false \
```

Comment:
again, remove prior_queue_opt from here, and check whether '--use-gpu false' is the appropriate option here.